src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -826,17 +827,33 @@ private void ProcessSettingsFrame(FrameHeader frameHeader, bool initialFrame = f
                             // We don't actually store this value; we always send frames of the minimum size (16K).
                             break;
 
+                        case SettingId.EnableConnect:
+                            if (settingValue == 1)
+                            {
+                                IsConnectEnabled.SetResult(true);
```

Comment:
The setting could be received multiple times.
```suggestion
                                IsConnectEnabled.TrySetResult(true);
```